server/scripts/20240825000000-invalidate-unfurls-cache.ts

```diff
@@ -0,0 +1,75 @@
+import RedisAdapter from "@server/storage/redis";
+import "./bootstrap";
+
+const redis = RedisAdapter.defaultClient;
+
+const Keys = {
+  Pattern: "unfurl:*",
+  Total: 0,
+  Error: 0,
+};
+
+const Page = {
+  Count: 0,
+  Limit: 1000,
+};
+
+const deleteKeys = async (keys: string[]): Promise<void> => {
+  Keys.Total += keys.length;
+
+  const pipeline = redis.pipeline();
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  keys.forEach((key: any) => pipeline.del(key));
+
+  const res = await pipeline.exec();
+
+  if (!res) {
+    Keys.Error += keys.length;
+    return;
+  }
+
+  res.forEach(([err]) => {
+    if (err) {
+      Keys.Error++;
+    }
+  });
+};
+
+const execute = async () => {
+  const stream = redis.scanStream({ match: Keys.Pattern, count: Page.Limit });
+
+  console.log("Invalidating unfurls cache...");
+
+  stream.on("data", (unfurlKeys) => {
+    if (!unfurlKeys.length) {
```

Comment:
The way redis stream works is,
Get the keys for the current page -> apply match filter -> return data to client.

It's sometimes possible that the returned data is empty but the stream iteration has not reached the end.